pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -51,6 +51,8 @@ func (c *Cluster) getLabels(daemonName string) map[string]string {
 	labels := opspec.PodLabels(AppName, c.Namespace, "mon", daemonName)
 	// Add "mon_cluster: <namespace>" for legacy
 	labels[monClusterAttr] = c.Namespace
+	labels["mon_canary"] = ""
```

Comment:
If you pass the `pvcName` and `canary` flags to this method, all the labels could be taken care of in this method.

Reply:
getLabel is only called in this file (not exportable). We only know if we are going to create a canary pod in the "mon.go" file. And i think that is simple to set with the rigth value this labels in the exact moment we need them.

Reply:
Instead of setting these values to blank and then overriding them later I really think we need to set them to the real values. We'll need to pass these values down from parent methods, even if we need to pass them through several parent methods. This will help resolve @BlaineEXE and my concerns that the correct labels are all set.